internal/reconciliations/istio/configuration.go

```diff
@@ -78,6 +83,108 @@ func EvaluateIstioCRChanges(istioCR operatorv1alpha1.Istio, istioTag string) (tr
 	return trigger, nil
 }
 
+func checkComponentsConfigChange(components *operatorv1alpha1.Components, components2 *operatorv1alpha1.Components) IstioCRChange {
+	if nilChange(components.Pilot, components2.Pilot) || (len(components.IngressGateways) != len(components2.IngressGateways)) {
+		return ConfigurationUpdate
+	}
+
+	if components.Pilot != nil {
+		if checkK8SConfigChange(components.Pilot.K8s, components2.Pilot.K8s) {
+			return ConfigurationUpdate
+		}
+	}
+
+	if len(components.IngressGateways) > 0 {
+		for i, ingressGateway := range components.IngressGateways {
+			if checkK8SConfigChange(ingressGateway.K8s, components2.IngressGateways[i].K8s) {
+				return ConfigurationUpdate
+			}
+		}
+	}
+
+	return NoChange
+}
+
+func checkK8SConfigChange(config operatorv1alpha1.KubernetesResourcesConfig, config2 operatorv1alpha1.KubernetesResourcesConfig) bool {
```

Comment:
I think this helps understanding what return values we can expect.
```suggestion
func hasK8SConfigChange(config operatorv1alpha1.KubernetesResourcesConfig, config2 operatorv1alpha1.KubernetesResourcesConfig) bool {
```